### PR TITLE
Remove `register_attr` feature

### LINKED ISSUE #66080: Tracking issue for `#![register_attr]`
This is a direct replacement for now removed `#![feature(custom_attribute)]` (https://github.com/rust-lang/rust/issues/29642), except it doesn't rely on implicit fallback from unresolved attributes to custom attributes (which was always hacky and is the primary reason for the removal of `custom_attribute`) and requires registering the attribute explicitly.

```rust
#![register_attr(my_attr)]

#[my_attr] // OK
fn main() {}
```

It's not yet clear whether this should go through stabilization or not.
It's quite possible that all the uses should migrate to `#![register_tool]` (https://github.com/rust-lang/rust/issues/66079) instead.

### PR DESCRIPTION
**Pull request by JohnTitor**
Closes #66080

Signed-off-by: Yuki Okushi <jtitor@2k36.org>

### CODE REVIEW

File: compiler/rustc_feature/src/removed.rs [hunk 1]
```diff
@@ -163,6 +163,9 @@ declare_features! (
     (removed, quad_precision_float, "1.0.0", None, None, None),
     (removed, quote, "1.33.0", Some(29601), None, None),
     (removed, reflect, "1.0.0", Some(27749), None, None),
+    /// Allows using the `#[register_attr]` attribute.
+    (removed, register_attr, "CURRENT_RUSTC_VERSION", Some(66080), None,
```
Comment (TaKO8Ki) [inline compiler/rustc_feature/src/removed.rs:167]: Could you add a "FIXME" or keep #66080 open until the next version is released?

Comment (JohnTitor) [reply compiler/rustc_feature/src/removed.rs:167]: I'm not sure what the intention is to keep the tracking issue open as the feature will be completely removed. Could you clarify?

Comment (TaKO8Ki) [reply compiler/rustc_feature/src/removed.rs:167]: I forgot to mention why. This is to remind us to change `CURRENT_RUSTC_VERSION` to a fixed version.

Comment (JohnTitor) [reply compiler/rustc_feature/src/removed.rs:167]: How does it help? According to https://github.com/rust-lang/rust/pull/100591 and https://forge.rust-lang.org/release/process.html#master-bootstrap-update-t-2-day-tuesday, `x.py` will replace it with an actual version number when preparing a new release, and I don't see any relation of the tracking issues at all. But it's kinda new stuff and I might be missing something, do you have any references for that?

Comment (TaKO8Ki) [reply compiler/rustc_feature/src/removed.rs:167]: Sorry. I did not fully understand the release process. Thanks for the explanation. It looks good.

Comment (est31) [reply compiler/rustc_feature/src/removed.rs:167]: @JohnTitor your understanding is correct.

**Pull request merged**